crates/ruff_python_trivia/src/tokenizer.rs

```diff
@@ -45,49 +45,27 @@ pub fn lines_before(offset: TextSize, code: &str) -> u32 {
 }
 
 /// Counts the empty lines between `offset` and the first non-whitespace character.
+#[allow(clippy::cast_possible_truncation)]
 pub fn lines_after(offset: TextSize, code: &str) -> u32 {
-    let mut cursor = Cursor::new(&code[offset.to_usize()..]);
-
-    let mut newlines = 0u32;
-    while let Some(c) = cursor.bump() {
-        match c {
-            '\n' => {
-                newlines += 1;
-            }
-            '\r' => {
-                cursor.eat_char('\n');
-                newlines += 1;
-            }
-            c if is_python_whitespace(c) => {
-                continue;
-            }
-            _ => {
-                break;
-            }
-        }
-    }
-
-    newlines
+    SimpleTokenizer::starts_at(offset, code)
+        .take_while(|token| {
+            token.kind == SimpleTokenKind::Newline || token.kind == SimpleTokenKind::Whitespace
+        })
+        .filter(|token| token.kind == SimpleTokenKind::Newline)
+        .count() as u32
 }
 
-/// Returns the position after skipping any trailing trivia up to, but not including the newline character.
-pub fn skip_trailing_trivia(offset: TextSize, code: &str) -> TextSize {
-    let tokenizer = SimpleTokenizer::starts_at(offset, code);
-
-    for token in tokenizer {
-        match token.kind() {
-            SimpleTokenKind::Whitespace
-            | SimpleTokenKind::Comment
-            | SimpleTokenKind::Continuation => {
-                // No op
-            }
-            _ => {
-                return token.start();
-            }
-        }
-    }
-
-    offset
+/// Counts the empty lines after `offset`, ignoring any trailing trivia on the same line as
+/// `offset`.
+#[allow(clippy::cast_possible_truncation)]
+pub fn lines_after_ignoring_trivia(offset: TextSize, code: &str) -> u32 {
+    SimpleTokenizer::starts_at(offset, code)
+        .skip_while(|token| token.kind != SimpleTokenKind::Newline && token.kind.is_trivia())
+        .take_while(|token| {
+            token.kind == SimpleTokenKind::Newline || token.kind == SimpleTokenKind::Whitespace
+        })
+        .filter(|token| token.kind == SimpleTokenKind::Newline)
+        .count() as u32
```

Comment:
Awkward to cast to `u32`, but the other methods in this file return `u32`, and the format builders that take the return value as argument also expect `u32` 🤷

Reply:
rust is build around assuming everything is `usize` and doesn't expect our u32 optimizations, so i'm fine if a bunch of casting

Reply:
It may be worth adding a `SAFETY: ...` comment explaining why casting down is fine.